api/database/dataRun.js

```diff
@@ -0,0 +1,212 @@
+	 *
+	 * 1. Fills an array of strings for each module-key pair
+     * 2. Compares the arrays for equality.
+	 *
+	 * @param {Array} newSchema - The data contents array of the deviceData
+     * object which contains modules and key-value pairs for data points.
+	 * @param {Array} oldSchema - The data contents array of an old deviceData
+     * object which contains modules and key-value pairs for data points.
+	 * @returns {{boolean}} - True if they have the same schemas (every
+     * module-key pair is identical)
+	 */
+    async compareModules (newSchema, oldSchema) {
+
+        let j;
+        // Check every sensor in the arrays
+        for (j = 0; j < newSchema.length; j++) {
+
+            let newArray = await this.modulekey_Array(j, newSchema);
+            let oldArray = await this.modulekey_Array(j, oldSchema);
+
+            // If schemas don't have same number of keys, they are not identical
+            if (newArray.length !== oldArray.length) {
+                console.log("Modules do not have the same number of keys at ", newSchema[j].module);
+                return false
+            }
+
+            // Check every data key for every module
+            let k;
+            for (k = 0; k < newArray.length; k++) {
+                const newSchemaSensor = newArray[k];
+                const oldSchemaSensor = oldArray[k];
+
+                if (newSchemaSensor !== oldSchemaSensor) {
+                    console.log("Module-key pair does not match!!!");
+                    console.log("New Sensor: ", newSchemaSensor);
+                    console.log("Old Sensor: ", oldSchemaSensor);
+                    return false; //need to go to the next data run
+                }      
+                // If they are equal, check the next module-key pair
+            }
+        }
+        return true;
+    }
+
+    /**
+	 * Declares and fills an array with strings for each module-key pair
+     * for a single module.
+	 *
+	 * 1. Counts the number of module-key pairs.
+     * 2. Initializes and fills the array with the module-key pairs.
+	 *
+	 * @param {number} j - The index corresponding to the module within
+     * the schema.
+	 * @param {Array} schema - The data contents array of a deviceData
+     * object which contains modules and key-value pairs for data points.
+	 * @returns {{Array}} - Returns the array of strings
+	 */
+    async modulekey_Array(j, schema) {
+
+        // Determine the number of keys in the data object
+        let numkeys = 0;
+        for (var key in schema[j].data) {
+            if (schema[j].data.hasOwnProperty(key)) {
+                numkeys++;
+            }
+        }
+        let modulekey_array = new Array(numkeys);
+        
+        // Fill array with strings for each module-key pair
+        let i = 0;
+        for (var key in schema[j].data) {
+            modulekey_array[i] = String(schema[j].module) + '-' + String(key);
+            i++;
+        }
```

Comment:
You check for hasOwnProperty in the first loop but not here. I also just found out you can use 

``` for (let key in Object.keys(schema[j].data) {

}```

to just loop over the object's own keys.